aip/0194.md

```diff
@@ -0,0 +1,92 @@
+---
+aip:
+  id: 194
+  state: reviewing
+  created: 2019-08-16
+permalink: /194
+redirect_from:
+  - /0194
+---
+
+# Retries for gRPC clients
+
+RPCs sometimes fail. When one does, the client performing the RPC needs to know
+whether it is safe to retry the operation. When status codes are used
+consistently across multiple APIs, clients can respond to failures
+appropriately.
+
+## Guidance
+
+Clients **should** retry requests for which repeated runs would not cause
+unintended state changes, which are non-transactional, and which are either
+unary or server streaming.
```

Comment:
Retrying server-streaming code is at least tricky at the moment. I'm not massively averse to keeping it in, but it's worth being aware that very language support it at the moment, IIRC.

Reply:
Done (removed)

Reply:
Done